src/iamserver/publicmessagehandler.cpp

```diff
@@ -0,0 +1,440 @@
+
+    mNodeChangedController.WriteToSteams(nodeInfo);
+}
+
+void PublicMessageHandler::OnNodeRemoved(const aos::String& id)
+{
+    LOG_DBG() << "On node removed: ID = " << id;
+}
+
+aos::Error PublicMessageHandler::SubjectsChanged(const aos::Array<aos::StaticString<aos::cSubjectIDLen>>& messages)
+{
+    LOG_DBG() << "Subjects changed: count = " << messages.Size();
+
+    iamproto::Subjects subjects;
+    ConvertToProto(messages, subjects);
+
+    mSubjectsChangedController.WriteToSteams(subjects);
+
+    return aos::ErrorEnum::eNone;
+}
+
+void PublicMessageHandler::Close()
+{
+    LOG_DBG() << "Close public message handler";
+
+    mNodeChangedController.Close();
+    mSubjectsChangedController.Close();
+}
+
+/***********************************************************************************************************************
+ * Protected
+ **********************************************************************************************************************/
+
+bool PublicMessageHandler::IsMainNode() const
+{
+    // Case-insensitive equality for strings
+    auto caseInsensitiveEqual = [](std::string_view a, std::string_view b) {
+        return std::equal(
+            a.begin(), a.end(), b.begin(), b.end(), [](char a, char b) { return std::tolower(a) == std::tolower(b); });
+    };
+
+    auto it = std::find_if(mNodeInfo.mAttrs.begin(), mNodeInfo.mAttrs.end(), [&](const auto& attr) {
+        return caseInsensitiveEqual(std::string_view(attr.mName.CStr(), attr.mName.Size()), cNodeTypeTag);
+    });
+
+    if (it != mNodeInfo.mAttrs.end()) {
+        return caseInsensitiveEqual(std::string_view(it->mValue.CStr(), it->mValue.Size()), cNodeTypeTagMainNodeValue);
+    }
+
+    // If attribute is not found, then it is the main node.
+    return true;
+}
+
+aos::Error PublicMessageHandler::SetNodeStatus(const aos::NodeStatus& status)
+{
+    auto err = mNodeInfoProvider->SetNodeStatus(status);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    err = mNodeManager->SetNodeStatus(mNodeInfo.mID, status);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+/***********************************************************************************************************************
+ * Private
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * IAMVersionService implementation
+ **********************************************************************************************************************/
+
+grpc::Status PublicMessageHandler::GetAPIVersion([[maybe_unused]] grpc::ServerContext* context,
+    [[maybe_unused]] const google::protobuf::Empty* request, iamanager::APIVersion* response)
+{
+    LOG_DBG() << "Public message handler. Process get API version";
```

Comment:
`Process get API version` is enough. Or add as additional parameter:
`Process get API version: handler=public`
or
`Process public get API version`

ditto for all the cases